tests/foreman/api/test_ansible.py

```diff
@@ -123,3 +129,56 @@ def test_positive_ansible_job_on_host(target_sat, module_org, rhel_contenthost):
     )
     result = target_sat.api.JobInvocation(id=job['id']).read()
     assert result.succeeded == 1
+
+    target_sat.api.Host(id=target_host.id).remove_ansible_role(data={'ansible_role_id': role_id})
+    host_roles = target_host.list_ansible_roles()
+    assert len(host_roles) == 0
+
+
+@pytest.mark.e2e
+@pytest.mark.stream
+@pytest.mark.tier2
+def test_add_and_remove_ansible_role_hostgroup(target_sat):
+    """
+    Test add and remove functionality for ansible roles in hostgroup via API
+
+    :id: 7672cf86-fa31-11ed-855a-0fd307d2d66b
+
+    :Steps:
+        1. Create a hostgroup
+        2. Sync few ansible roles
+        3. Assign a few ansible roles with the host group
+        4. Add some ansible role with the host group
+        5. Remove the added ansible roles from the host group
+
+    :expectedresults:
+        1. Ansible role assign/add/remove functionality should work as expected in API
+
+    :BZ: 2164400
+    """
+    ROLE_NAMES = [
+        'theforeman.foreman_scap_client',
+        'redhat.satellite.hostgroups',
+        'RedHatInsights.insights-client',
+    ]
+    ROLES = []
+    hg_name = gen_string('alpha')
+    hg = target_sat.api.HostGroup(name=hg_name).create()
+    assert hg.name == hg_name
```

Comment:
Not really needed to HG creation in othercomponent tests! HG is part of your setup so, just create hg without separate `hg_name` var.

Reply:
Didn't get you here, hg_name var isn't required in the test but required for setup, I think its better to validate if hg is created before moving ahead with test

Reply:
We should never assert test setups, infact we assume it passes. There are tests to validate if hg is being created with the name. Need not to really waste 2 extra lines on it. You can `assert hg` but not really required to check its name is same.

Its optional comment, leaving upto you to accept/ignore.

Reply:
Yes, I get it, I removed it, thanks